Source/Core/DolphinUWP/App.cpp

```diff
@@ -0,0 +1,331 @@
+  }
+
+  void Load(hstring const&) {}
+
+  void Uninitialize() {}
+
+  void Run()
+  {
+    while (m_running.IsSet())
+    {
+      if (m_shutdown_requested.TestAndClear())
+      {
+        const auto ios = IOS::HLE::GetIOS();
+        const auto stm = ios ? ios->GetDeviceByName("/dev/stm/eventhook") : nullptr;
+        if (!m_tried_graceful_shutdown.IsSet() && stm &&
+            std::static_pointer_cast<IOS::HLE::STMEventHookDevice>(stm)->HasHookInstalled())
+        {
+          ProcessorInterface::PowerButton_Tap();
+          m_tried_graceful_shutdown.Set();
+        }
+        else
+        {
+          m_running.Clear();
+        }
+      }
+
+      ::Core::HostDispatchJobs();
+      CoreWindow::GetForCurrentThread().Dispatcher().ProcessEvents(
+          CoreProcessEventsOption::ProcessAllIfPresent);
+
+      std::this_thread::sleep_for(std::chrono::milliseconds(1));
+    }
+  }
+
+  winrt::fire_and_forget InitializeDolphinFromFile(std::string path)
+  {
+    if (path.empty())
+    {
+      FileOpenPicker openPicker;
+      openPicker.ViewMode(PickerViewMode::List);
+      openPicker.SuggestedStartLocation(PickerLocationId::ComputerFolder);
+      openPicker.FileTypeFilter().Append(L".iso");
+      openPicker.FileTypeFilter().Append(L".ciso");
+      openPicker.FileTypeFilter().Append(L".rvz");
+      openPicker.FileTypeFilter().Append(L".wbfs");
+      openPicker.FileTypeFilter().Append(L".gcm");
+      openPicker.FileTypeFilter().Append(L".gcz");
+
+      auto file = co_await openPicker.PickSingleFileAsync();
+      if (file)
+      {
+        path = winrt::to_string(file.Path().data());
+      }
+    }
+
+    CoreWindow window = CoreWindow::GetForCurrentThread();
+    void* abi = winrt::get_abi(window);
+
+    WindowSystemInfo wsi;
+    wsi.type = WindowSystemType::UWP;
+    wsi.render_window = abi;
+    wsi.render_surface = abi;
+    wsi.render_width = window.Bounds().Width;
+    wsi.render_height = window.Bounds().Height;
+
+    auto navigation = winrt::Windows::UI::Core::SystemNavigationManager::GetForCurrentView();
+
+    // UWP on Xbox One triggers a back request whenever the B button is pressed
+    // which can result in the app being suspended if unhandled
+    navigation.BackRequested([](const winrt::Windows::Foundation::IInspectable&,
+                                const BackRequestedEventArgs& args) { args.Handled(true); });
+
+    GAMING_DEVICE_MODEL_INFORMATION info = {};
+    GetGamingDeviceModelInformation(&info);
+    if (info.vendorId == GAMING_DEVICE_VENDOR_ID_MICROSOFT)
+    {
+      HdmiDisplayInformation hdi = HdmiDisplayInformation::GetForCurrentView();
+      if (hdi)
+      {
+        wsi.render_width = hdi.GetCurrentDisplayMode().ResolutionWidthInRawPixels();
+        wsi.render_height = hdi.GetCurrentDisplayMode().ResolutionHeightInRawPixels();
+      }
+    }
+
+    std::unique_ptr<BootParameters> boot =
+        BootParameters::GenerateFromFile(path, BootSessionData("", DeleteSavestateAfterBoot::No));
```

Comment:
Can be simplified to:

```suggestion
    std::unique_ptr<BootParameters> boot = BootParameters::GenerateFromFile(path);
```